File: solace/cmd/cooldown/Slash.java
package solace.cmd.cooldown;

import java.util.*;
import solace.game.*;
import solace.net.*;
import java.io.*;
import solace.util.*;

/**
 * "Slash" is a cooldown action that does 150 potency damage to the attacker's
 * target. If executed after the "flurry of blows" attack it has a combo potency
 * of 250.
 * @author Ryan Sandor Richards
 */
public class Slash extends CooldownCommand {
  public Slash(Player p) {
    super("slash", p);
    setCooldownDuration(CooldownCommand.GLOBAL_COOLDOWN);
    setInitiatesCombat(true);
    setBasePotency(150);
    setComboPotency(225);
    setCombosWith("flurry");
    addResourceCost(new SpCost(4));
  }

  public boolean execute(int level, Player target) {
    return executeAttack(target);
  }
}

File: game/skills/one-handed/cooldowns/slash.js
'use strict';

/**
 * "Slash" is a cooldown action that does 125 potency damage to the attacker's
 * target. If executed after the "flurry of blows" attack it has a combo potency
 * of 200.
 * @author Ryan Sandor Richards
 */
Commands.addCooldown('slash', {
  cooldownDuration: Commands.GLOBAL_COOLDOWN,
  initiatesCombat: true,
  basePotency: 125,
  comboPotency: 200,
  combosWith: 'flurry',
  spCost: 4,
  run: function (player, target, level, cooldown) {
    return cooldown.executeAttack(player, target);
  }
});
